bundled/tool/lsp_server.py

```diff
@@ -126,7 +126,15 @@ def _get_severity(
     code: str, code_type: str, severity: Dict[str, str]
 ) -> lsp.DiagnosticSeverity:
     """Converts severity provided by linter to LSP specific value."""
-    value = severity.get(code, None) or severity.get(code_type, "Error")
+    value = None
+
+    while code and value is None:
+        value = severity.get(code, None)
+        code = code[:-1]
+
+    if value is None:
+        severity.get(code_type, "Error")
+
```

Comment:
This implementation could be redone as:
```python
def _get_severity(code: str, code_type: str, severity: Dict[str, str]) -> lsp.DiagnosticSeverity:
    """Converts severity provided by linter to LSP specific value."""

    try:
        while code:
            if code in severity:
                return lsp.DiagnosticSeverity[severity[code]]
            code = code[:-1]

        if code_type in severity:
            return lsp.DiagnosticSeverity[severity[code_type]]
    except KeyError:
        pass

    return lsp.DiagnosticSeverity.Error
```

Reply:
This will require double lookup on the severity dictionary. If you don't mind that, I'll accept the changes.

Reply:
Yeah I agree. Restored to what you had with the fix for `value` with `code_type`: 
```python
def _get_severity(code: str, code_type: str, severity: Dict[str, str]) -> lsp.DiagnosticSeverity:
    """Converts severity provided by linter to LSP specific value."""
    value = None
    while code and value is None:
        value = severity.get(code, None)
        code = code[:-1]

    if value is None:
        value = severity.get(code_type, "Error")

    try:
        return lsp.DiagnosticSeverity[value]
    except KeyError:
        pass

    return lsp.DiagnosticSeverity.Error
```

Reply:
Great, thanks! The fix was applied in 8714960.